File: erpnext/hr/report/programacion_supervisores/programacion_supervisores.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt


import requests

def execute(filters=None):
	columns, data = [], []
	supervisor = requests.post("https://horario-salida-qa-erp.shalom.com.pe/api/reporte/reporte_supervisores", json = filters)
	supervisor = supervisor.json()

	if "data" in supervisor:
		return get_columns(), supervisor.get("data")


def get_columns():
	columns = [
		{
			'label': "NOMBRE SUPERVISOR",
			'fieldname': 'nombre_supervisor',
			'fieldtype': 'Data',
			'width': 500
		},
		{
			'label': "TOTAL DE PROGRAMACIONES",
			'fieldname': 'total_programaciones',
			'fieldtype': 'Float',
			'width': 180
		},
		{
			'label': "INFORME ADJUNTO",
			'fieldname': 'informe_adjunto',
			'fieldtype': 'Float',
			'width': 180
		},
		{
			'label': "INFORMES APROBADOS",
			'fieldname': 'aprobados',
			'fieldtype': 'Float',
			'width': 180
		},
		{
			'label': "% DE EFICIENCIA",
			'fieldname': 'eficiencia',
			'fieldtype': 'Data',
			'width': 170
		}
	]

	return columns

File: erpnext/hr/report/programacion_supervisores/programacion_supervisores.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */
let today = frappe.datetime.get_today();
frappe.query_reports["Programacion Supervisores"] = {

	"filters": [
		{
			"fieldname":"date_init",
			"label": __("From"),
			"fieldtype": "Date",
			"default": frappe.datetime.month_start(frappe.datetime.get_today()),
			"width": "100px"
		},
		{
			"fieldname":"date_end",
			"label": __("Hasta"),
			"fieldtype": "Date",
			"default": frappe.datetime.month_end(frappe.datetime.get_today()),
			"width": "100px"
		}
	]
};